sdk/tables/Azure.Data.Tables/src/TableConnectionString.cs

```diff
@@ -269,6 +271,84 @@ static Uri CreateUri(string endpoint, string sasToken)
             return false;
         }
 
+        private const string TableNameGroup = "tableName";
+        private static readonly Regex s_tableUriTableNameRegex = new Regex(@$"(Tables\('(?<{TableNameGroup}>\w+)'\))");
```

Comment:
add timeout

Reply:
Compiled? Would indexOf be simpler?

Reply:
I meant to make it compiled, thanks for the call out. 

I considered doing an indexOf for `Tables('`, but this seems more straight forward than parsing out that, and the trailing `'`.